pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1643,6 +1661,18 @@ func appNetworkDoInactivateUnderlayNetwork(
 		startDnsmasq(bridgeName)
 	}
 	netstatus.RemoveVif(log, ulStatus.Vif)
+	if netstatus.Type == types.NetworkInstanceTypeSwitch {
+		if ulStatus.AccessVlanID <= 1 {
+			netstatus.NumTrunkPorts--
+		} else {
+			netstatus.VlanMap[ulStatus.AccessVlanID]++
+			if netstatus.VlanMap[ulStatus.AccessVlanID] > 0 {
+				netstatus.VlanMap[ulStatus.AccessVlanID]--
+			} else {
+				delete(netstatus.VlanMap, ulStatus.AccessVlanID)
+			}
```

Comment:
Why do you add and subtract?  Isn't the intent to subtrace and then delete when it reaches zero?

Reply:
My bad. Coding error. Will correct it.